tracetools_trace/tracetools_trace/tools/lttng_impl.py

```diff
@@ -242,7 +246,12 @@ def setup(
     if not session_name:
         raise RuntimeError('empty session name')
     # Resolve full tracing directory path
-    full_path = os.path.join(base_path, session_name)
+    if snapshot_session and dual_session:
+        full_path = os.path.join(base_path, session_name.removesuffix('-snapshot'), 'snapshot')
+    elif dual_session:
+        full_path = os.path.join(base_path, session_name.removesuffix('-runtime'), 'runtime')
```

Comment:
As mentioned in another thread, I would use `RUNTIME_SESSION_SUFFIX` and `SNAPSHOT_SESSION_SUFFIX` here